java/arcs/core/data/proto/ParticleSpecProtoDecoder.kt

```diff
@@ -35,8 +37,15 @@ fun HandleConnectionSpecProto.decode() = HandleConnectionSpec(
 )
 
 /** Converts a [ParticleSpecProto] to the corresponding [ParticleSpec] instance. */
-fun ParticleSpecProto.decode() = ParticleSpec(
-    name = getName(),
-    connections = getConnectionsList().map { it.decode() },
-    location = getLocation()
-)
+fun ParticleSpecProto.decode(): Result<ParticleSpec> = resultOf {
+    val connections = mutableMapOf<String, HandleConnectionSpec>()
+    for (connection in getConnectionsList()) {
```

Comment:
can this be: `connectionsList.forEach {`?

(Java Bean-style getters/setters in java classes - like those generated from protos - can usually be used like properties/fields in kotlin)

Reply:
Yep, I just discovered accidentally that proto fields can be accessed as properties in Kotlin. Felt like Christmas. :D